File: sample-videochat-webrtc/sample-videochat-webrtc/Components/Call/CallHelper/MediaListener.m
//
//  MediaListener.m
//  sample-videochat-webrtc
//
//  Created by Injoit on 06.10.2021.
//  Copyright © 2021 QuickBlox Team. All rights reserved.
//

#import "MediaListener.h"

@implementation MediaListener

- (void)controller:(nonnull SessionsController *)controller didBroadcastMediaType:(MediaType)mediaType enabled:(BOOL)enabled {
    switch (mediaType) {
        case MediaTypeAudio:
            if (self.onAudio) {
                self.onAudio(enabled);
            }
            break;
        case MediaTypeVideo:
            if (self.onVideo) {
                self.onVideo(enabled);
            }
            break;
        case MediaTypeSharing:
            if (self.onSharing) {
                self.onSharing(enabled);
            }
            break;
    }
}

- (void)controller:(nonnull SessionsController *)controller didReceivedRemoteVideoTrack:(nonnull QBRTCVideoTrack *)videoTrack fromUser:(nonnull NSNumber *)userID {
    if (self.onReceivedRemoteVideoTrack) {
        self.onReceivedRemoteVideoTrack(videoTrack, userID);
    }
}

@end

File: sample-videochat-webrtc-swift/sample-videochat-webrtc-swift/Components/Calls/CallHelper/MediaListener.swift
//
//  MediaListener.swift
//  sample-videochat-webrtc-swift
//
//  Created by Injoit on 07.12.2021.
//  Copyright © 2021 QuickBlox. All rights reserved.
//

import Foundation
import QuickbloxWebRTC

typealias ReceivedRemoteVideoTrackHandler = ( _ videoTrack: QBRTCVideoTrack, _ userID: NSNumber) -> Void
typealias BroadcastHandler = ( _ enabled: Bool) -> Void

class MediaListener {
    //MARK: - Properties
    var onReceivedRemoteVideoTrack: ReceivedRemoteVideoTrackHandler?
    var onAudio: BroadcastHandler?
    var onVideo: BroadcastHandler?
    var onSharing: BroadcastHandler?
}

extension MediaListener: SessionsMediaListenerDelegate {
    func controller(_ controller: SessionsController, didBroadcastMediaType mediaType: MediaType, enabled: Bool) {
        
        switch mediaType {
        case .audio: onAudio?(enabled)
        case .video:onVideo?(enabled)
        case .sharing: onSharing?(enabled)
        }
    }
    
    func controller(_ controller: SessionsController, didReceivedRemoteVideoTrack videoTrack: QBRTCVideoTrack, fromUser userID: NSNumber) {
        onReceivedRemoteVideoTrack?(videoTrack, userID)
    }
}
